src/utils/appData.ts

```diff
@@ -21,12 +26,57 @@ async function getValidator(): Promise<{ ajv: Ajv; validate: ValidateFunction }>
   return { ajv, validate }
 }
 
-export async function validateAppDataDocument(appDataDocument: any): Promise<ValidationResult> {
+function buildCidInstance(hash: string) {
+  const cidVersion = 0x1 //.toString(16) //cidv1
+  const codec = 0x70 //.toString(16) //dag-pb
+  const type = 0x12 //.toString(16) //sha2-256
+  const length = 32 //.toString(16) //256 bits
+  const _hash = hash.replace(/(^0x)/, '')
+
+  const hexHash = fromHexString(_hash)
+
+  if (!hexHash) return
+
+  const uint8array = Uint8Array.from([cidVersion, codec, type, length, ...hexHash])
+
+  return new CID(uint8array)
+}
+
+async function loadIpfsFromCid(cid: string) {
+  const response = await fetch(`${DEFAULT_IPFS_URI}/${cid}`)
+
+  return await response.json()
+}
+
+export async function validateAppDataDocument(appDataDocument: unknown): Promise<ValidationResult> {
   const { ajv, validate } = await getValidator()
-  const result = !!(await validate(appDataDocument))
+  const result = !!validate(appDataDocument)
 
   return {
     result,
     errors: ajv.errors ?? undefined,
   }
 }
+
+export async function decodeAppData(hash: string) {
```

Comment:
I know you are simply porting my initial code, but could you also improve it?

It was meant as a quick hack so I did not really put much thought into types and so on. Would be nice to have it done properly here :)